lib/proxy/request.rb

```diff
@@ -63,9 +63,14 @@ def http
     private
 
     def http_init
-      http             = Net::HTTP.new(uri.host, uri.port)
-      http.use_ssl     = uri.scheme == 'https'
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = uri.scheme == 'https'
       http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+      http.open_timeout = Proxy::SETTINGS.foreman_http_timeout
+      http.ssl_timeout = Proxy::SETTINGS.foreman_http_timeout
+      http.continue_timeout = Proxy::SETTINGS.foreman_http_timeout
```

Comment:
Reading https://ruby-doc.org/stdlib-2.7.1/libdoc/net/http/rdoc/Net/HTTP.html#attribute-i-continue_timeout
> Seconds to wait for 100 Continue response. If the HTTP object does not receive a response in this many seconds it sends the request body. The default value is nil.

It doesn't say what nil actually means. Does that disable the timeout?

I really doubt we use HTTP 100 Continue so should we modify the default behavior? I'm leaning to be safe and not set it.
```suggestion
```

Reply:
Okay makes sense, I had no idea honestly what that is, now it makes sense, HTTP 100.